api/api/services/sendgrid_mail.py

```diff
@@ -1,72 +1,90 @@
-            '[voron] new email from voron.sh!',
-        )
+    def extract_first_name(self, email: str):
+        """Extract first name from email using regex"""
+        match = re.match(r"^[^._@]+", email)
+        return match.group() if match else ""
 
-    def set_template_data(self, data: dict):
-        """Send back a dictionnary with the data to be substituted"""
-        self.mail.dynamic_template_data = {
-            **data
-        }
-
-    def set_template_id(self, template_id: str):
-        """Set template ID before sending email"""
-        self.mail.template_id = template_id
+    def set_template_data(self, data: dict, recipient_email: str):
+        """Set dynamic data for the template"""
+        first_name = self.extract_first_name(recipient_email)
+        data["username"] = first_name
+        self.mail.dynamic_template_data = data
 
     def set_attachment(self, filename: str, content: bytes, **kwargs):
         """Set file attachment"""
         file_type = kwargs.get("type", "application/pdf")
-        content_id = kwargs.get("content_id", "Pentest_report.pdf")
+        content_id = kwargs.get("content_id", "attachment.pdf")
 
         attachment = Attachment()
-        attachment.file_content = base64.b64encode(content).decode('utf-8')
+        attachment.file_content = FileContent(
+            base64.b64encode(content).decode('utf-8'))
         attachment.file_type = FileType(file_type)
         attachment.file_name = FileName(filename)
         attachment.disposition = Disposition('attachment')
         attachment.content_id = ContentId(content_id)
-
         self.mail.attachment = attachment
 
     def send(self):
-        """send email"""
-
+        """Send email"""
         if os.environ.get('TEST', '0') == '1':
-            logging.warning('[DEBUG] Email has hit .send() function')
+            self.logger.warning('[DEBUG] Email has hit .send() function')
             return
 
         try:
             return self.sendgrid_client.send(self.mail)
         except Exception as error:
             self.logger.exception(error)
-            raise error
+            raise
+
+
+def get_recipients():
+    url = f'https://{os.environ["DOMAIN_NAME"]}/api/mailing-list'
+    response = requests.get(url)
+    return response.json().get("results", [])
+
+
+if __name__ == "__main__":
+    recipients = get_recipients()
+
+    for recipient_data in recipients:
+        email_address = recipient_data.get("email")
+        email_client = SendgridClient(recipient=email_address)
+        email_client.set_template_data({
+            "text": "Welcome to our service!",
+            "profile": "https://img.freepik.com/photos-gratuite/surface-abstraite-textures-mur-pierre-beton-blanc_74190-8189.jpg?size=626&ext=jpg&ga=GA1.1.1546980028.1703030400&semt=ais",
+            "email": email_address
+        }, recipient_email=email_address)
+        response = email_client.send()
+        print(f"Email sent to {email_address}. Response:",
```

Comment:
use a warning here instead of a print, otherwise it won't be seen from docker logs